src/main-gcu.cpp

```diff
@@ -849,16 +875,26 @@ static errr Term_xtra_gcu_react(void)
 
 #ifdef A_COLOR
 
-    int i;
-
-    /* Cannot handle color redefinition */
-    if (!can_fix_color)
-        return (0);
-
-    /* Set the colors */
-    for (i = 0; i < 16; i++) {
-        /* Set one color (note scaling) */
-        init_color(i, angband_color_table[i][1] * 1000 / 255, angband_color_table[i][2] * 1000 / 255, angband_color_table[i][3] * 1000 / 255);
+    if (COLORS == 256 || COLORS == 88)
+    {
+        /* If we have more than 16 colors, find the best matches. These numbers
+        * correspond to xterm/rxvt's builtin color numbers--they do not
+        * correspond to curses' constants OR with curses' color pairs.
+        *
+        * XTerm has 216 (6*6*6) RGB colors, with each RGB setting 0-5.
+        * RXVT has 64 (4*4*4) RGB colors, with each RGB setting 0-3.
+        *
+        * Both also have the basic 16 ANSI colors, plus some extra grayscale
+        * colors which we do not use.
+        */
+        int i;
```

Comment:
This variable can be put inside a for-loop.

Reply:
OK, fixed. I'm not sure what you mean by changing to C++; this file looks exactly like the main-gcu.c files present in other variants of the same generation, I suspect it hasn't been modified since the entire codebase was still in C.